services/provider/server/storageclassrequest.go

```diff
@@ -0,0 +1,169 @@
+package server
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	controllers "github.com/red-hat-storage/ocs-operator/controllers/storageconsumer"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+type storageClassRequestManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newStorageClassRequestManager(ctx context.Context, cl client.Client, namespace string) (*storageClassRequestManager, error) {
+	return &storageClassRequestManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+// getStorageClassRequestName generates a name for a StorageClassRequest resource.
+func getStorageClassRequestName(consumerUUID, storageClassRequestName string) string {
+	var s struct {
+		StorageConsumerUUID     string `json:"storageConsumerUUID"`
+		StorageClassRequestName string `json:"storageClassRequestName"`
+	}
+	s.StorageConsumerUUID = consumerUUID
+	s.StorageClassRequestName = storageClassRequestName
+
+	requestName, err := json.Marshal(s)
+	if err != nil {
+		klog.Errorf("failed to marshal a name for a storage class request based on %v. %v", s, err)
+		panic("failed to marshal storage class request name")
+	}
+	name := md5.Sum([]byte(requestName))
+	// The name of the StorageClassRequest is the MD5 hash of the JSON
+	// representation of the StorageClassRequest name and storageConsumer UUID.
+	return fmt.Sprintf("storageclassrequest-%s", hex.EncodeToString(name[:16]))
+}
+
+// Create creates a new StorageClassRequest resource and returns the StorageClassRequest ID.
+func (s *storageClassRequestManager) Create(
+	ctx context.Context,
+	consumer *ocsv1alpha1.StorageConsumer,
+	storageClassRequestName,
+	requestType,
+	encryptionMethod,
+	storageProfile string,
+) error {
+	consumerUUID := string(consumer.GetUID())
+	generatedRequestName := getStorageClassRequestName(consumerUUID, storageClassRequestName)
+
+	storageClassRequestObj := &ocsv1alpha1.StorageClassRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generatedRequestName,
+			Namespace: s.namespace,
+			Labels: map[string]string{
+				controllers.ConsumerUUIDLabel: consumerUUID,
+				storageClassRequestName:       storageClassRequestName,
```

Comment:
@subhamkrai This should be `storageClassClaimNameLabel` we are adding name for both the label key and label value.

Reply:
Let's prioritize it and get it in both 4.13 and 4.12